### PR TITLE
Unify the system peripheral

### PR DESCRIPTION
**Pull request by MabezDev**
Whilst the PCR, SYSTEM and DPORT peripherals are different, we currently use them all in the same way. This PR unifies the peripheral name in the hal to `SYSTEM`. The idea is that they all do the same sort of thing, so we can collect them under the same name, and later down the line we can begin to expose differences under an extended API.

The benefits to this are imo quite big, the examples now are all identical, which makes things easier for esp-wifi, and paves a path towards the multichip hal.

Why not do this in the PAC? Imo the pac should be as close to the hardware as possible, and the HAL is where we should have abstractions such as this.

### Must

- [x] The code compiles without `errors` or `warnings`.
- [x] All examples work.
- [x] `cargo fmt` was run.
- [ ] Your changes were added to the `CHANGELOG.md` in the proper section.
- [ ] You updated existing examples or added examples (if applicable).
- [ ] Added examples are checked in CI

### Nice to have

- [x] You add a description of your work to this PR.
- [x] You added proper docs for your newly added features and code.

### CODE REVIEW

**Review by bjoernQ — APPROVED** [review]
I love it!

Just one idea:
Wouldn't it make sense to have something like `RADIO => virtual` instead of `RADIO => false` to make it clearer what this does / means

**Comment by MabezDev** [comment]
> Wouldn't it make sense to have something like RADIO => virtual instead of RADIO => false to make it clearer what this does / means

Nice idea! I like that much better.

File: esp-hal-common/src/embassy/executor/xtensa/thread.rs [hunk 1]
```diff
@@ -6,9 +6,9 @@ use core::{
 
 use embassy_executor::{raw, Spawner};
 #[cfg(all(dport, multi_core))]
-use peripherals::DPORT as SystemPeripheral;
+use peripherals::DPORT as SYSTEM;
```
Comment (bugadani) [inline esp-hal-common/src/embassy/executor/xtensa/thread.rs:9]: Shouldn't this go away, too?

File: esp-hal-common/src/soc/esp32/peripherals.rs [hunk 2]
```diff
@@ -29,49 +29,49 @@ pub use pac::Interrupt;
...
-    UART1 => true,
-    UART2 => true,
-    UHCI0 => true,
-    UHCI1 => true,
-    RADIO => false,
-    PSRAM => false,
+    AES => AES,
+    APB_CTRL => APB_CTRL,
+    BB => BB,
+    SYSTEM => DPORT,
```
[the remark this answers is not in the document]
Comment (MabezDev) [reply esp-hal-common/src/soc/esp32/peripherals.rs:36]: Good point! I added a comment which should hopefully be enough to differentiate between the others. If we need to do this sort of thing often I will expand the functionality of the macro at a later date.

File: esp-hal-common/src/peripheral.rs [hunk 3]
```diff
@@ -353,7 +329,7 @@ mod peripheral_macros {
 
             impl crate::peripheral::sealed::Sealed for $name {}
         };
-        ($(#[$cfg:meta])? $name:ident => false) => {
+        ($(#[$cfg:meta])? $name:ident => $base:ident) => {
```
Comment (jessebraham) [inline esp-hal-common/src/peripheral.rs:332]: I feel like swapping these parameters is more intuitive (you star with the base and go toward the final type name). I know this has been commented on already, and I won't block merging based on this alone, but this is pretty confusing in its current form, IMO.

**Review by jessebraham — APPROVED** [review]
Left one comment, but overall LGTM. Thanks for taking care of this, I think it's a nice improvement.

**Pull request merged**